### PR TITLE
Add porkbun support

### LINKED ISSUE #3: Update Porkbun dns record if mismatched

### PR DESCRIPTION
**Pull request by chrispinkney**
## Issue
closes #2

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
This PR adds Porkbun support and re-structures the project for Namecheap support.

## Testing the PR
1. Input porkbun tokens in `cmd/hosts.env`.
2. Run the program and assert that specified record is updated.

## Checklist

- [ ] **Quality**: This PR builds and passes the respective npm run test and works locally
- [ ] **Tests**: This PR includes thorough tests
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: constants/constants.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package constants
+
+const (
+	Filename           = "hosts.env"
+	EditByNameTypeURL  = "https://api.porkbun.com/api/json/v3/dns/editByNameType/"
+	RetrieveByNameType = "https://api.porkbun.com/api/json/v3/dns/retrieveByNameType/"
```
Comment (manekenpix) [inline constants/constants.go]: Should we have these in an env file? If any of the providers change their URLs we'll have to make a change in the repo, but if we have these in an env file, it'll just be a matter of updating your local env file.

Comment (chrispinkney) [reply constants/constants.go]: Agreed, maybe even a separate file in the future or something if this gets too large. 👍

File: services/ipservice.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	"io"
+	"net/http"
+	"strings"
+)
+
+func GetPublicIP() (string, error) {
+	services := []string{
+		"https://api.ipify.org",
+		"https://ifconfig.me",
+		"https://ipinfo.io/ip",
```
Comment (manekenpix) [inline services/ipservice.go]: We should probably have these in the `.env` file?

Comment (chrispinkney) [reply services/ipservice.go]: I want to tidy this up a bit in the future, maybe with a few more `IP_SERVICE` additions in the `.env` file or something and iterate through them but for now I think it's ok- LMK

**Review by manekenpix — APPROVED** [review]

**Pull request merged**